tests/rdma/run.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python3
+"""
+==========================================================================
+run.py - script for test client for Valkey Over RDMA (Linux only)
+--------------------------------------------------------------------------
+Copyright (C) 2024  zhenwei pi <pizhenwei@bytedance.com>
+
+This work is licensed under BSD 3-Clause, License 1 of the COPYING file in
+the top-level directory.
+==========================================================================
+"""
+import os
+import subprocess
+import netifaces
+import time
+import argparse
+
+def build_program():
+    valkeydir = os.path.dirname(os.path.abspath(__file__)) + "/../.."
+    cmd = "make -C " + valkeydir + "/tests/rdma"
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+    if p.wait():
+        print("Valkey Over RDMA build rdma-test [FAILED]")
+        return 1
+
+    print("Valkey Over RDMA build rdma-test program [OK]")
+    return 0
+
+
+# iterate /sys/class/infiniband, find any usable RDMA device, and return IPv4 address
+def find_rdma_dev():
+    # Ex, /sys/class/infiniband/mlx5_0
+    # Ex, /sys/class/infiniband/rxe_eth0
+    # Ex, /sys/class/infiniband/siw_eth0
+    ibclass = "/sys/class/infiniband/"
+    try:
+        for dev in os.listdir(ibclass):
+            # Ex, /sys/class/infiniband/rxe_eth0/ports/1/gid_attrs/ndevs/0
+            netdev = ibclass + dev + "/ports/1/gid_attrs/ndevs/0"
+            with open(netdev) as fp:
+                addrs = netifaces.ifaddresses(fp.readline().strip("\n"))
+                if netifaces.AF_INET not in addrs:
+                    continue
+                ipaddr = addrs[netifaces.AF_INET][0]["addr"]
+                print("Valkey Over RDMA test prepare " + dev + " <" + ipaddr  + "> [OK]")
+                return ipaddr
+    except os.error:
+        return None
+
+    return None
+
+
+def test_rdma(ipaddr):
+    valkeydir = os.path.dirname(os.path.abspath(__file__)) + "/../.."
+    retval = 0
+
+    # step 1, prepare test directory
+    tmpdir = valkeydir + "/tests/rdma/tmp"
+    subprocess.Popen("mkdir -p " + tmpdir, shell=True).wait()
```

Comment:
Do we use this tmp dir for anything?

Reply:
> Do we use this tmp dir for anything?

Yes, server gets started with `--dir /PATH/TO/TMP`, and `dump.rdb` is generated on `BGSAVE` command.